save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/codeeditor/SandboxCodeEditorComponent.kt

```diff
@@ -0,0 +1,204 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS", "HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.frontend.components.basic.codeeditor
+
+import com.saveourtool.save.frontend.components.basic.codeeditor.FileType.CODE
+import com.saveourtool.save.frontend.components.basic.codeeditor.FileType.Companion.getTypedOption
+import com.saveourtool.save.frontend.components.basic.codeeditor.FileType.SAVE_TOML
+import com.saveourtool.save.frontend.components.basic.codeeditor.FileType.SETUP_SH
+import com.saveourtool.save.frontend.components.views.sandboxApiUrl
+import com.saveourtool.save.frontend.externals.reactace.AceModes
+import com.saveourtool.save.frontend.externals.reactace.AceThemes
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.info.UserInfo
+
+import csstype.ClassName
+import react.FC
+import react.Props
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.h6
+import react.useState
+
+import kotlinx.browser.window
+import kotlinx.coroutines.await
+
+private const val DEFAULT_EDITOR_MESSAGE = "Select one of the files above to start editing it!"
+
+/**
+ * CodeEditor component
+ */
+val sandboxCodeEditorComponent = sandboxCodeEditorComponent()
+
+/**
+ * SandboxCodeEditor functional component [Props]
+ */
+external interface SandboxCodeEditorComponentProps : Props {
+    /**
+     * Title of an editor
+     */
+    var editorTitle: String?
+
+    /**
+     * Action to run execution
+     */
+    var doRunExecution: () -> Unit
+
+    /**
+     * Action to reload debug info
+     */
+    var doResultReload: () -> Unit
+
+    /**
+     * todo: will be removed soon
```

Comment:
A link to a relevant Github issue would be helpful if we forget to remove it soon :)

Reply:
Well, it will be removed right now as #1310 is merged